File: HelpSoftware/Help/Help.Web/Web/src/app/EF/PinWallComment.ts
import { User } from "./EF";

export class PinWallComment {
  ID_PinWallComment: number;
  Author: User;
  User: User;
  CommentTime: Date;
  CommentTitle: string;
  CommentText: string;
  IsPublic: boolean;
  IsDeleted: boolean;
  IsAnonymous: boolean;

  deserialize(input): PinWallComment {
    this.ID_PinWallComment = input.ID_PinWallComment;
    this.Author = new User().deserialize(input.Author);
    this.User = new User().deserialize(input.User);
    this.CommentTime = input.CommentTime;
    this.CommentTitle = input.CommentTitle;
    this.CommentText = input.CommentText;
    this.IsPublic = input.IsPublic;
    this.IsDeleted = input.IsDeleted;
    this.IsAnonymous = input.IsAnonymous;
    return this;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/PinWallComment.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class PinWallComment : IHelpEntity
  {
    [Key]
    public virtual int ID_PinWallComment { get; set; }

    public virtual User Author { get; set; }

    public virtual User User { get; set; }

    public virtual DateTime CommentTime { get; set; }

    public virtual string CommentTitle { get; set; }

    public virtual string CommentText { get; set; }

    public virtual bool IsPublic { get; set; }

    public virtual bool IsDeleted { get; set; }

    public virtual bool IsAnonymous { get; set; }

    [Timestamp]
    public virtual byte[] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "PinWallComment";
    }

    public int GetID()
    {
      return ID_PinWallComment;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}